mantidimaging/gui/widgets/zslider/zslider.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2023 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+from __future__ import annotations
+
+from PyQt5.QtCore import pyqtSignal, Qt
+from PyQt5.QtWidgets import QGraphicsSceneMouseEvent
+from pyqtgraph import PlotItem, InfiniteLine
+
+
+class ZSlider(PlotItem):
```

Comment:
We should probably have a class doc string here to describe PlotItem subclass item.